src/lib/util/VersionUpgradeManager.ts

```diff
@@ -43,6 +46,23 @@ upgradeScriptsByVersion.set('v3.0.0', (config: ConfigContent): Promise<void> =>
 	}
 	return Promise.resolve();
 });
+upgradeScriptsByVersion.set('v3.6.0', async (config: ConfigContent): Promise<void> => {
+	// v3.6.0 is the first v3.x version that's GA instead of a pilot release.
+	// To provide pilot users with a continuity of behavior, we want to overwrite
+	// the existing GA config with the pilot config, if one exists and it is deemed
+	// appropriate to do so.
+	const fh: FileHandler = new FileHandler();
+	const pilotConfigPath = path.join(Controller.getSfdxScannerPath(), CONFIG_PILOT_FILE);
+	if (await fh.exists(pilotConfigPath)) {
+		// If the pilot config exists, read it in.
+		const pilotConfig: ConfigContent = JSON.parse(await fh.readFile(pilotConfigPath)) as ConfigContent;
+		// If the pilot config is for a later plug-in version than the current GA config, replace
+		// the GA config's `engines` property with that of the pilot config.
+		if (!config.currentVersion || semver.lt(config.currentVersion, pilotConfig.currentVersion)) {
```

Comment:
If `Config.ts` generated a config from the pilot, then its `currentVersion` is guaranteed to be <3.6.0, so this script will run.
In this case, copying the contents is technically harmless, but redundant. This `if`-check lets us skip that case.